src/invalidations.jl

```diff
@@ -255,6 +255,41 @@ else
     new_backedge_table() = Dict{Tuple{Int32,UInt64},Tuple{Any,Vector{Any}}}()
 end
 
+"""
+    report_invalidations(;
+        job_name::String = "",
+        invalidations,
+        n_rows::Int = 10,
+        process_filename::Function = x -> x,
+    )
+
+Print a tabular summary of invalidations given:
+
+ - `job_name::String` the name of the job
+
+An `@info` message also prints some information, for example
+if the table has been truncated (upon request).
+
+```julia
+import SnoopCompileCore
+invalidations = SnoopCompileCore.@snoopr begin
+
+    # load packages & do representative work
```

Comment:
```suggestion
    # load packages & define any additional methods
```
"Doing work" cannot invalidate; invalidations are caused by new method definitions, which includes loading packages.

Reply:
Good point. I think I phrased it that way to be safe. I think I could use some clarification. If someone's "workload" was, for example:

```julia
function do_work(A)
    foo(a::Array) = a .+ 1
    x = zero(eltype(A))
    for i in 1:2
        x += sum(foo(A))
    end
    return x
end
```
Can `foo` in this example possibly invalidate methods? If so, then I think users may find `do representative work` to be a safety net to catch unexpected invalidations in real world applications.

Though, it's probably important to note that the "amount of work" (e.g., the size of `A` in this example) has no impact on the invalidations.

Overall, I think `define any additional methods` is more precise, it just requires that users think carefully about where / when they may be defining methods, which I think is fair. I'll commit this change, but I'd appreciate if you can chime in regarding my example. Thanks!

Reply:
> If someone's "workload" was

That's not a workload, that's a method definition. `do_work(rand(5))` is a workload. Perhaps we need to clarify that workload means *running* code.

> Can foo in this example possibly invalidate methods?

Inner methods can't invalidate. But `do_work` could invalidate, if there are other methods.